File: app/src/main/java/com/ranzan/mojopizzaclone/ImageSlider/ImageSliderAdapter.java
package com.ranzan.mojopizzaclone.ImageSlider;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import androidx.viewpager2.widget.ViewPager2;

import com.makeramen.roundedimageview.RoundedImageView;
import com.ranzan.mojopizzaclone.R;

import java.util.List;
import java.util.zip.Inflater;

public class ImageSliderAdapter extends RecyclerView.Adapter<ImageSliderAdapter.SlideHolder> {
    private List<ImageSliderClass> sliderItem;
    private ViewPager2 viewPager2;

    public ImageSliderAdapter(List<ImageSliderClass> sliderItem, ViewPager2 viewPager2) {
        this.sliderItem = sliderItem;
        this.viewPager2 = viewPager2;
    }

    @NonNull
    @Override
    public SlideHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new SlideHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.slider_item,parent,false));
    }

    @Override
    public void onBindViewHolder(@NonNull SlideHolder holder, int position) {
        ImageSliderClass imageSliderClass= sliderItem.get(position);
        holder.setimage(imageSliderClass);
    }
    private Runnable runnable = new Runnable() {
        @Override
        public void run() {
            sliderItem.addAll(sliderItem);
            notifyDataSetChanged();
        }
    };

    @Override
    public int getItemCount() {
        return sliderItem.size();
    }

    public class SlideHolder extends RecyclerView.ViewHolder{
        private RoundedImageView roundedImageView;
        public SlideHolder(@NonNull View itemView) {
            super(itemView);
            roundedImageView=itemView.findViewById(R.id.image_slider_item);
        }
        void setimage(ImageSliderClass imageSliderClass)
        {
            roundedImageView.setImageResource(imageSliderClass.getImg());
        }
    }
}

File: app/src/main/java/com/ranzan/mojopizzaclone/ImageSlider/ImageSliderAdapter.kt
package com.ranzan.mojopizzaclone.ImageSlider

import com.ranzan.mojopizzaclone.ImageSlider.ImageSliderClass
import androidx.viewpager2.widget.ViewPager2
import androidx.recyclerview.widget.RecyclerView
import com.ranzan.mojopizzaclone.ImageSlider.ImageSliderAdapter.SlideHolder
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import com.ranzan.mojopizzaclone.R
import com.makeramen.roundedimageview.RoundedImageView

class ImageSliderAdapter(
    private val sliderItem: List<ImageSliderClass>,
    private val viewPager2: ViewPager2
) : RecyclerView.Adapter<SlideHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SlideHolder {
        return SlideHolder(
            LayoutInflater.from(parent.context).inflate(R.layout.slider_item, parent, false)
        )
    }

    override fun onBindViewHolder(holder: SlideHolder, position: Int) {
        val imageSliderClass = sliderItem[position]
        holder.setimage(imageSliderClass)
    }

    override fun getItemCount(): Int {
        return sliderItem.size
    }

    inner class SlideHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val roundedImageView: RoundedImageView
        fun setimage(imageSliderClass: ImageSliderClass) {
            roundedImageView.setImageResource(imageSliderClass.img)
        }

        init {
            roundedImageView = itemView.findViewById(R.id.image_slider_item)
        }
    }
}